src/ai/OpenRouter.js

```diff
@@ -8,17 +8,38 @@ export const OpenRouter = class extends OpenAI {
   constructor(options) {
     options.baseURL ||= 'https://openrouter.ai/api/v1';
     super(options);
+    const models = this.model.split(';');
+    if (models.length > 1) {
+        this.model = models[0];
+        this.fallbacks = models.slice(1);
+    }
   }
 
   async init() {
     if (this.didInit) {
       return;
     }
 
-    const parts = this.model.split('/');
-    const data = await getModelData(parts[0], parts[1], this.cache);
+    // Get primary model information
+    let parts = this.model.split('/');
+    let data = await getModelData(parts[0], parts[1], this.cache);
     this.maxTokens = data.maxTokens;
     this.pricing = data.pricing;
+
+    // Get fallback model information
+    if (this.fallbacks) {
+      const datas = await Promise.all(this.fallbacks.map(async fallback => {
+        const parts = fallback.split('/');
+        const data = await getModelData(parts[0], parts[1], this.cache);
```

Comment:
nit: just `return getModelData(parts[0], parts[1], this.cache);`.

async functions return promises, so await the result is redundant